### PR TITLE
refactor: generic support && time task persistance

### PR DESCRIPTION
**Pull request by icceey**
- upgrade to go 1.18, support generic
- `/run_after` : time task persistance
- `/hitokoto` : fix extra `《` and `》`
- fix telebot middleware, now `/fake_ban` `no_sticker` `recorder` can work

### CODE REVIEW

File: util/timer/timer.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
+package timer
+
+import (
+	"sort"
+	"sync"
+)
+
+// TODO: use heap when package `heap` is ready
```
Comment (icceey) [inline util/timer/timer.go:8]: use 了吗，timer还要吗

**Review by icceey — COMMENTED** [review]
viewed

File: util/timer/timer.go (hunk 1 above)
Comment (hugefiver) [reply util/timer/timer.go:8]: 不要了吧

Comment (icceey) [reply util/timer/timer.go:8]: 那要不删了吧，还是先这样放着

**Review by hugefiver — APPROVED** [review]

**Pull request merged**